onecodex/models/sample.py

```diff
@@ -24,6 +24,11 @@ def where(cls, *filters, **keyword_filters):
         public = keyword_filters.get('public', False)
         instances_route = 'instances' if not public else 'instances_public'
         limit = keyword_filters.get('limit', None if not public else 1000)
+        tag = keyword_filters.pop('tag', None)
```

Comment:
So, we actually support this in the API and shouldn't need to get all samples and filter in memory. Here's an example public URL:

```
https://app.onecodex.com/api/v1/samples?where={"tags": {"$contains": {"$ref": "/api/v1/tags/dbcf5b98bca54a16"}}}
```

Consequently, I think the best way to implement this is to fetch the tag by name if we have a non-UUID (and maybe have some kind of `is_uuid()` function to skip this check) and then include it in the `.where` query directly.

Note we also implement a custom `$containsany` and `$containsall` selector in our backend that allows passing an array of tags and doing more complex intersects. We should look at the performance of that query code before leaning on it too heavily, but those may be useful too.

Reply:
OK, this was a lot simpler. Changed as requested.